File: leetcode/1.go
func twoSum(nums []int, target int) []int {
	hashMap := make(map[int] int)
	// 创建一个空 hash 表
	for i, num := range nums {
			if j, ok := hashMap[target - num]; ok {
					return []int{i, j}
			}
			hashMap[num] = i
	}
	return nil
}

File: data-structures-algorithms/leetcode/dart/1.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    // map diff
    Map<int, int> map = {};
    for (int i = 0; i < nums.length; i++) {
      if (map.containsKey(nums[i])) {
        return [map[nums[i]]!, i];
      } else {
        map[target - nums[i]] = i;
      }
    }
    return [];
  }
}
